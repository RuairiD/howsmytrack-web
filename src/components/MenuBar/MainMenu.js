import React from "react";

import { Div } from "lemon-reset";

import LoggedInMenu from "./LoggedInMenu";
import LoggedOutMenu from "./LoggedOutMenu";
import SocialLinksMenu from "./SocialLinksMenu";

const MainMenu = ({
    isMobile,
    onMenuClick,
    username,
    rating,
    notifications,
}) => (
    <Div>
        {username && (
            <LoggedInMenu
                onMenuClick={onMenuClick}
                username={username}
                rating={rating}
                notifications={notifications}
            />
        )}
        {!username && (
            <LoggedOutMenu
                onMenuClick={onMenuClick}
            />
        )}
        <SocialLinksMenu
            isMobile={isMobile}
            onMenuClick={onMenuClick}
        />
    </Div>
);

export default MainMenu;
