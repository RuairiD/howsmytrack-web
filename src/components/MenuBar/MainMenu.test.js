import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MainMenu from "./MainMenu";

describe("MainMenu", () => {
    it("renders LoggedInMenu and SocialLinksMenu when a username is provided, implying that the user is logged in", () => {
        const wrapper = shallow(
            <MainMenu
                onMenuClick={() => {}}
                username="username"
                rating={4.321}
                notifications={2}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders LoggedInMenu and SocialLinksMenu when a username is provided, implying that the user is logged in", () => {
        const wrapper = shallow(
            <MainMenu
                onMenuClick={() => {}}
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
