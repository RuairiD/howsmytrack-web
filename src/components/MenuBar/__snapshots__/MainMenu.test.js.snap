// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu renders LoggedInMenu and SocialLinksMenu when a username is provided, implying that the user is logged in 1`] = `
<Div
  className=""
>
  <LoggedInMenu
    notifications={2}
    onMenuClick={[Function]}
    rating={4.321}
    username="username"
  />
  <SocialLinksMenu
    onMenuClick={[Function]}
  />
</Div>
`;

exports[`MainMenu renders LoggedInMenu and SocialLinksMenu when a username is provided, implying that the user is logged in 2`] = `
<Div
  className=""
>
  <LoggedOutMenu
    onMenuClick={[Function]}
  />
  <SocialLinksMenu
    onMenuClick={[Function]}
  />
</Div>
`;
